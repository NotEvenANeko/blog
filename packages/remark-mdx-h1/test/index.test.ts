import { describe, expect, it } from "vitest";
import * as fs from "node:fs/promises";
import * as path from "node:path";
import { compile } from "@mdx-js/mdx";

import remarkMdxH1 from "../src";

describe("remark-mdx-h1", () => {
  it("should extract h1 and remove it", async () => {
    const md = await fs.readFile(
      path.join(import.meta.dirname, "./fixtures/test-case-1.md"),
      "utf-8"
    );

    const result = (
      await compile(md, {
        remarkPlugins: [remarkMdxH1],
        rehypePlugins: [],
      })
    ).toString();

    expect(result).toMatchSnapshot();
  });

  it("should not extract sub heading if not exists", async () => {
    const md = await fs.readFile(
      path.join(import.meta.dirname, "./fixtures/test-case-2.md"),
      "utf-8"
    );

    const result = (
      await compile(md, {
        remarkPlugins: [remarkMdxH1],
        rehypePlugins: [],
      })
    ).toString();

    expect(result).toMatchSnapshot();
  });

  it("should not remove h1 if opt out of remove", async () => {
    const md = await fs.readFile(
      path.join(import.meta.dirname, "./fixtures/test-case-1.md"),
      "utf-8"
    );

    const result = (
      await compile(md, {
        remarkPlugins: [[remarkMdxH1, { removeHeading: false }]],
        rehypePlugins: [],
      })
    ).toString();

    expect(result).toMatchSnapshot();
  });
});
