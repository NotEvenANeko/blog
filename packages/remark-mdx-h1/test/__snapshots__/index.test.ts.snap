// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`remark-mdx-h1 > should extract h1 and remove it 1`] = `
"import {Fragment as _Fragment, jsx as _jsx, jsxs as _jsxs} from "react/jsx-runtime";
export const h1 = {
  "heading": "I'm Heading 1",
  "subHeading": "This is sub heading"
};
function _createMdxContent(props) {
  const _components = {
    h2: "h2",
    p: "p",
    ...props.components
  };
  return _jsxs(_Fragment, {
    children: [_jsx(_components.p, {
      children: "This is not"
    }), "\\n", _jsx(_components.h2, {
      children: "Heading 2"
    })]
  });
}
export default function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = props.components || ({});
  return MDXLayout ? _jsx(MDXLayout, {
    ...props,
    children: _jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
"
`;

exports[`remark-mdx-h1 > should not extract sub heading if not exists 1`] = `
"import {jsx as _jsx} from "react/jsx-runtime";
export const h1 = {
  "heading": "I'm H1",
  "subHeading": undefined
};
function _createMdxContent(props) {
  const _components = {
    h2: "h2",
    ...props.components
  };
  return _jsx(_components.h2, {
    children: "There's no sub heading"
  });
}
export default function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = props.components || ({});
  return MDXLayout ? _jsx(MDXLayout, {
    ...props,
    children: _jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
"
`;

exports[`remark-mdx-h1 > should not remove h1 if opt out of remove 1`] = `
"import {Fragment as _Fragment, jsx as _jsx, jsxs as _jsxs} from "react/jsx-runtime";
export const h1 = {
  "heading": "I'm Heading 1",
  "subHeading": "This is sub heading"
};
function _createMdxContent(props) {
  const _components = {
    h1: "h1",
    h2: "h2",
    p: "p",
    ...props.components
  };
  return _jsxs(_Fragment, {
    children: [_jsx(_components.h1, {
      children: "I'm Heading 1"
    }), "\\n", _jsx(_components.p, {
      children: "This is sub heading"
    }), "\\n", _jsx(_components.p, {
      children: "This is not"
    }), "\\n", _jsx(_components.h2, {
      children: "Heading 2"
    })]
  });
}
export default function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = props.components || ({});
  return MDXLayout ? _jsx(MDXLayout, {
    ...props,
    children: _jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
"
`;
